server/migrations/20210730210120-add-fileOperations.js

```diff
@@ -0,0 +1,65 @@
+'use strict';
+
+module.exports = {
+  up: async (queryInterface, Sequelize) => {
+    await queryInterface.createTable("file_operations",{
+      id: {
+        type: Sequelize.UUID,
+        allowNull: false,
+        primaryKey: true,
+      },
+      state: {
+        type: Sequelize.ENUM("creating", "uploading", "complete", "error","expired"),
+        allowNull: false,
+      },
+      type: {
+        type: Sequelize.ENUM("import", "export"),
+        allowNull: false,
+      },
+      key: {
+        type: Sequelize.STRING,
+      },
+      url: {
+        type: Sequelize.STRING,
+      },
+      size: {
+        type: Sequelize.BIGINT,
+        allowNull: false,
+      },
+      userId: {
+        type: Sequelize.UUID,       
+        allowNull: false,
+        references: {
+          model: "users"
+        }
+      },
+      collectionId: {
+        type: Sequelize.UUID,       
+        references: {
+          model: "collections"
+        }
+      },
+      teamId: {
+        type: Sequelize.UUID,
+        allowNull: false,
+        references: {
+          model: "teams"
+        }
+      },
+      createdAt: {
+        type: Sequelize.DATE,
+        allowNull: false,
+      },
+      updatedAt: {
+        type: Sequelize.DATE,
+        allowNull: false,
+      },
+    })
+    await queryInterface.addIndex('file_operations', ["state","type"]);
```

Comment:
This adds a single compound-index which would require "state" to be in the query, I don't think that's what you meant to do?

```suggestion
    await queryInterface.addIndex('file_operations', ["state"]);
    await queryInterface.addIndex('file_operations', ["type"]);
```

If not `removeIndex` below will also need changing

Reply:
We use `type` and `state` attributes while querying. I thought a compound index would be must faster than having a single index.

Suppose we are not going with compound indexes. I think the `type` index is all that we need.  

```    
where: {
  type: "export",
  createdAt: {
    [Op.lt]: subDays(new Date(), 30),
  },
  state: {
    [Op.ne]: "expired",
  },
 },
 ```

Reply:
Noted.